### PR TITLE
AGP bumps and build improvements

### PR DESCRIPTION
**Pull request by romtsn**
_#skip-changelog_

## :pencil: Checklist

- [ ] I reviewed the submitted code
- [ ] I added tests to verify the changes
- [ ] I updated the docs if needed
- [ ] No breaking changes

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.3.3-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.3.3-bin.zip
```
Comment (romtsn) [inline gradle/wrapper/gradle-wrapper.properties:3]: was told by the gradle team that it's better to use -bin rather than -all

Comment (cortinico) [reply gradle/wrapper/gradle-wrapper.properties:3]: Why this? IIRC the `-bin` doesn't come with sources. Therefore, you will miss on some features (i.e. cmd+b inside the `.gradle` files and so on).

Comment (marandaneto) [reply gradle/wrapper/gradle-wrapper.properties:3]: I recall this being an issue as well.

File: buildSrc/src/main/java/Dependencies.kt [hunk 2]
```diff
@@ -11,7 +11,7 @@ object BuildPluginsVersion {
 
 object LibsVersion {
     const val JUNIT = "4.13.2"
-    const val ASM = "9.2"
+    const val ASM = "7.0" // compatibility matrix -> https://developer.android.com/reference/tools/gradle-api/7.1/com/android/build/api/instrumentation/InstrumentationContext#apiversion
```
Comment (marandaneto) [inline buildSrc/src/main/java/Dependencies.kt:14]: If we're using AGP 7.1.x, it should be ASM9, no?

Comment (marandaneto) [reply buildSrc/src/main/java/Dependencies.kt:14]: If this is a problem, we can add the ASM version to the CI matrix, and read it as `VERSION_AGP`

Comment (romtsn) [reply buildSrc/src/main/java/Dependencies.kt:14]: so, we actually get the ASM version from the AGP, when the plugin is running on the clients projects (hence, the compileOnly change for ASM). So no matter what we use here, it will be overridden by the gradle's dependency resolution to what AGP is using.

I just defined a min. supported version here (as the min support AGP version is 7.0.0 -> ASM 7.0). So it's similar to our kotlin compatibility approach.

Basically, our sample apps would be built against what ASM version AGP is using underneath, so the current CI matrix that we have for AGP versions is sufficient.

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (romtsn) [reply gradle/wrapper/gradle-wrapper.properties:3]: ah, right, I guess they meant this probably for the apps that usually don't do much gradle logic/plugins, but for us it'd make sense to have sources indeed.

File: buildSrc/src/main/java/Dependencies.kt (hunk 2 above)
Comment (marandaneto) [reply buildSrc/src/main/java/Dependencies.kt:14]: gotcha, makes sense.

**Review by cortinico — APPROVED** [review]

**Review by marandaneto — APPROVED** [review]

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (cortinico) [reply gradle/wrapper/gradle-wrapper.properties]: Yup 👍  thanks for clarifying
